### PR TITLE
[googlemaps] info-window JSDoc, tests & events

### PR DESCRIPTION
**Pull request by demensky**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/maps/documentation/javascript/reference/info-window
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/googlemaps/reference/info-window.d.ts [hunk 1]
```diff
@@ -1,78 +1,208 @@
...
+         * @see {@link InfoWindowHandlerMap#domready domready} event
+         * @see {@link InfoWindowHandlerMap#position_changed position_changed} event
+         * @see {@link InfoWindowHandlerMap#zindex_changed zindex_changed} event
+         */
+        addListener<N extends keyof InfoWindowHandlerMap>(
+            eventName: N,
+            handler: MVCEventHandler<this, InfoWindowHandlerMap[N]>,
+        ): MapsEventListener;
+
+        addListener(eventName: string, handler: MVCEventHandler<this, unknown[]>): MapsEventListener;
```
Comment (simonhaenisch) [inline types/googlemaps/reference/info-window.d.ts:73]: What is this general overload good for? For other event types of `MVCObject`? Couldn't the first one also include the keys of `MVCObject` events?

File: types/googlemaps/reference/info-window.d.ts [hunk 2]
```diff
@@ -1,78 +1,208 @@
 declare namespace google.maps {
+    interface InfoWindowHandlerMap {
+        /**
+         * This event is fired when the close button was clicked.
+         * @see {@link https://developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow.closeclick Maps JavaScript API}
+         * @see {@link InfoWindow#close}
+         */
+        closeclick: [];
```
Comment (simonhaenisch) [inline types/googlemaps/reference/info-window.d.ts:8]: Why are these arrays?

Comment (demensky) [reply types/googlemaps/reference/info-window.d.ts:8]: Some events ([example](https://developers.google.com/maps/documentation/javascript/reference/event#MVCArray.remove_at)) have more than one argument. I began to apply a similar approach on the [`Map` events](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4b51763cdda43057b20a89a00b61a1f34d6fe4d5/types/googlemaps/reference/map.d.ts#L2-L133). This was done for consistency.

File: types/googlemaps/reference/info-window.d.ts (hunk 1 above)
Comment (demensky) [reply types/googlemaps/reference/info-window.d.ts:73]: In the [`Map`object](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4b51763cdda43057b20a89a00b61a1f34d6fe4d5/types/googlemaps/reference/map.d.ts#L168-L169), I used this so as not to break backward compatibility. [`MVCObject`](https://developers.google.com/maps/documentation/javascript/reference/event#MVCObject) has no events.

I can replace it with:

```suggestion
        /** @deprecated */
        addListener(eventName: string, handler: MVCEventHandler<this, any[]>): MapsEventListener;
```

Do you propose to remove it altogether?

**Review by simonhaenisch — APPROVED** [review]

**Review by simonhaenisch — APPROVED** [review]

**Pull request merged**